actioncable/lib/rails/generators/channel/templates/assets/channel.js

```diff
@@ -0,0 +1,18 @@
+App.<%= class_name.underscore %> = App.cable.subscriptions.create("<%= class_name %>Channel", {
+  connected: function() {
+    // Called when the subscription is ready for use on the server
+  },
+
+  disconnected: function() {
+    // Called when the subscription has been terminated by the server
+  },
+
+  received: function(data) {
+    // Called when there's incoming data on the websocket for this channel
+  }<%= actions.count > 0 ? ",\n" : '' %>
```

Comment:
Alternatively:

```
  received: function(data) {
    // Called when there's incoming data on the websocket for this channel
<% actions.each do |action| -%>
  },

  <%= action %>: function() {
    return this.perform('<%= action %>');
<% end -%>
  }
});
```

Not sure which I prefer, tbh... this option has less code / conditionals, but has a slightly less straight-forward mapping to the output.